src/devices/things/ITemperatureSensor.cs

```diff
@@ -0,0 +1,14 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Iot.Device
+{
+    /// <summary>
+    /// Abstraction for temperature
+    /// </summary>
+    public interface ITemperatureSensor
+    {
+        float Temperature { get; }
```

Comment:
Doesn't it need to specify method name as well? like
```C#
float GetTemperature();
```

Reply:
If there is a property, I don't think we need a function especially if the function does not bring anything different.

Reply:
@ZhangGaoxing property getter is technically a method (at least on IL level)

Reply:
Indeed property getter is a method but properties syntax suggests operations that are very fast, basically returning already stored variable. If such call is intended to call sensor it should be explicit method: GetTemperature suggesting a bit more complex operation.

Reply:
mmm I'm not sure that is necessarily a true every time. While most of the time this is true because properties would only provide the value of a private member, I do believe we have some examples in the framework where properties have a bit more logic behind them. 

That said, I do have as feedback that we should be explicit with the units that these properties return, so this should be something like: `TemperatureInCelcius`. That way, bindings that consume types that implement this interface would know exactly what to expect out of the values provided by these properties.

Reply:
It should be explicit for what is not international standard. Acutally Temperature is in Celcius (in theory, it should even been in Kelving as it's the real international one :-)) 
Where it's more problematic, it's distance sensors like ultrasound. In the international metric system, the unit is the meter. So if there is a default value, it should be in meter. Now, most of those sensors in Arduino and Python are usually returning centimeters. So in this case, yes, it would make sense to have Something like DistanceInCentimeter because it's not meter. And it's usually centimeter because it's the default unit and does not require a float.
but if the "Distance" returns meter, then no need to add anything.
Other option I proposed up is to have Something called "UnitToMeter" which is then a float/double to give the conversion rate to the main international unit. So in case of "Distance" returning centimeter, the "UnitToMeter" would then be 0.01.

Reply:
@arkadiuszwojcik this may or may not be fast - depends on the implementation - how fast is fast? In many of the sensors you can do hundreds of reads per second. Just `Temperature` is much shorter and more intuitive